### PR TITLE
add raster type conversion

### PR DESCRIPTION
**Pull request by jdroenner**

### CODE REVIEW

**Comment by jdroenner** [comment]
[Raster Type Conversion Operator](https://trello.com/c/GWZCtJy4/963-raster-type-conversion-operator)

File: src/operators/rastertypeconversion.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# RasterTypeConversion
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:1]: I guess you also need to add that here: https://github.com/geo-engine/docs/blob/main/src/SUMMARY.md.

File: src/operators/rastertypeconversion.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+The `RasterTypeConversion` operator allows change the data type of raster data.
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:3]:
```suggestion
The `RasterTypeConversion` operator allows changing the data type of raster data.
```

I recommend installing the Grammarly plugin for VSCode.

File: src/operators/rastertypeconversion.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+It transforms all pixels into the new data type.
+
+1. The operator allows loss of precision e.g. converting a `f32` value of "3.1" to a `u8` will return a value of "3".
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:6]:
```suggestion
1. Applying the operator could lead to a loss of precision, e.g., converting a `F32` value of `3.1` to a `U8` will return a value of `3`.
```

File: src/operators/rastertypeconversion.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+2. If the old value is not valid in the new type it will clip at the value range of the new type.
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:8]: Maybe add an example here as well for non computer scientists :smile:

File: src/operators/rastertypeconversion.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+## Parameters
+
+| Parameter | Type                     | Description                  | Example Value         |
+| --------- | ------------------------ | ---------------------------- | --------------------- |
+| `outputDataType` | [`RasterDataType`]| the output type              | "U8"                  |
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:14]:
```suggestion
| Parameter        | Type               | Description     | Example Value |
| ---------------- | ------------------ | ----------------| ------------- |
| `outputDataType` | [`RasterDataType`] | the output type | `U8`          |
```

File: src/operators/rastertypeconversion.md [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+The `RasterTypeConversion` operator expects either one _raster_ input.
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:16]:
```suggestion
The `RasterTypeConversion` operator expects exactly one _raster_ input.
```

File: src/operators/rastertypeconversion.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+| Parameter | Type                         |
+| --------- | ---------------------------- |
+| `source`  | `SingleRasterSource` |
```
Comment (ChristianBeilschmidt) [inline src/operators/rastertypeconversion.md:20]:
```suggestion
| Parameter | Type                 |
| --------- | -------------------- |
| `source`  | `SingleRasterSource` |
```

**Review by ChristianBeilschmidt — CHANGES_REQUESTED** [review]

File: src/operators/rastertypeconversion.md (hunk 2 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 1 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md:1]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 2 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: the plugin does not work :(

File: src/operators/rastertypeconversion.md (hunk 3 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 4 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 5 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 6 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

File: src/operators/rastertypeconversion.md (hunk 7 above)
Comment (jdroenner) [reply src/operators/rastertypeconversion.md]: [f61834f](https://github.com/geo-engine/docs/pull/29/commits/f61834f48103b630a6d859ffcffe669c977bceb0)

**Review by ChristianBeilschmidt — APPROVED** [review]

**Comment by ChristianBeilschmidt** [comment]
bors merge

**Pull request merged**